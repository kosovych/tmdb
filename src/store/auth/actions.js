import {
  LOGIN_REQUEST, LOGIN_FAILS, LOGIN_SUCCESS, LOGIN_SUBMIT
} from './types'

export const loginSubmit = ({ username, password }) => ({
  type: LOGIN_SUBMIT,
  username,
  password
})

export const loginFails = (errorTxt, loading) => ({
  type: LOGIN_FAILS,
  errorTxt,
  loading
})

export const loginSuccess = (sessionID, loading) => ({
  type: LOGIN_SUCCESS,
  sessionID,
  loading
})

export const loginRequest = (loading, error) => ({
  type: LOGIN_REQUEST,
  loading,
  error
})
