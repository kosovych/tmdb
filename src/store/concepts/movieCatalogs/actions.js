import {
  GET_MOVIES, REQUEST_MOVIES_START, REQUEST_MOVIES_SUCCESS, REQUEST_MOVIES_ERROR, SET_SEARCH
} from './types'

export const getMovies = params => ({
  type: GET_MOVIES,
  params
})

export const requestMoviesStart = endpoint => ({
  type: REQUEST_MOVIES_START,
  endpoint
})

export const requestMoviesSuccess = (endpoint, payload) => ({
  type: REQUEST_MOVIES_SUCCESS,
  endpoint,
  payload
})

export const requestMoviesError = (endpoint, error) => ({
  type: REQUEST_MOVIES_ERROR,
  endpoint,
  error
})

export const setSearch = (endpoint, searchQuery) => ({
  type: SET_SEARCH,
  endpoint,
  searchQuery
})
