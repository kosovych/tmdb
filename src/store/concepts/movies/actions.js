import {
  GET_MOVIES, REQUEST_MOVIES_START, REQUEST_MOVIES_SUCCESS, REQUEST_MOVIES_ERROR
} from './types'

export const getMovies = (endpoint, url) => ({
  type: GET_MOVIES,
  endpoint,
  url
})

export const requestMoviesStart = endpoint => ({
  type: REQUEST_MOVIES_START,
  endpoint
})

export const requestMoviesSuccess = (endpoint, payload) => ({
  type: REQUEST_MOVIES_SUCCESS,
  endpoint,
  payload
})

export const requestMoviesError = (endpoint, error) => ({
  type: REQUEST_MOVIES_ERROR,
  endpoint,
  error
})
