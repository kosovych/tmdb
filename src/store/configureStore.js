import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'
import axios from '../axios';

import logic from './logics';
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {
    axios
  }

  const logicMiddleware = createLogicMiddleware(logic, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
