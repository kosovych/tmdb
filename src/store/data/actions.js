import {
  STORE_DATA,
  STORE_MOVIE_INFO
} from './types'

export const storeData = (endpoint, payload) => ({
  type: STORE_DATA,
  endpoint,
  payload
})

export const storeMovieInfo = (movieId, payload) => ({
  type: STORE_MOVIE_INFO,
  movieId,
  payload
})
