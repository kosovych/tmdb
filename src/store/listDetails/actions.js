import {
  GET_LIST_DETAILS,
  REQUEST_START,
  REQUEST_SUCCESS,
  REQUEST_ERROR,
  REMOVE_MOVIE,
  REMOVE_USER_LIST
} from './types'

export const getListDetails = listId => ({
  type: GET_LIST_DETAILS,
  listId
})

export const requestStart = () => ({
  type: REQUEST_START
})

export const requestSuccess = (name, entries) => ({
  type: REQUEST_SUCCESS,
  name,
  entries
})

export const requestError = () => ({
  type: REQUEST_ERROR
})

export const removeMovie = (movieId, listId) => ({
  type: REMOVE_MOVIE,
  movieId,
  listId
})

export const removeUserList = (listId, history) => ({
  type: REMOVE_USER_LIST,
  listId,
  history
})
