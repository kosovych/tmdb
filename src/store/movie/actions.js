import {
  GET_MOVIE_INFO,
  REQUEST_MOVIE_INFO_START,
  REQUEST_MOVIE_INFO_SUCCESS,
  REQUEST_MOVIE_INFO_ERROR,
  GET_MOVIE_IMAGES,
  REQUEST_MOVIE_IMAGES_START,
  REQUEST_MOVIE_IMAGES_SUCCESS,
  REQUEST_MOVIE_IMAGES_ERROR,
  GET_MOVIE_CREDITS,
  REQUEST_MOVIE_CREDITS_START,
  REQUEST_MOVIE_CREDITS_ERROR,
  REQUEST_MOVIE_CREDITS_SUCCESS
} from './types'

export const getMovie = movieId => ({
  type: GET_MOVIE_INFO,
  movieId
})

export const requestMovieInfoStart = () => ({
  type: REQUEST_MOVIE_INFO_START
})

export const requestMovieInfoSuccess = info => ({
  type: REQUEST_MOVIE_INFO_SUCCESS,
  info
})

export const requestMovieInfoError = () => ({
  type: REQUEST_MOVIE_INFO_ERROR
})

export const getMovieImages = movieId => ({
  type: GET_MOVIE_IMAGES,
  movieId
})

export const requestMovieImagesStart = () => ({
  type: REQUEST_MOVIE_IMAGES_START
})

export const requestMovieImagesSuccess = images => ({
  type: REQUEST_MOVIE_IMAGES_SUCCESS,
  images
})

export const requestMovieImagesError = () => ({
  type: REQUEST_MOVIE_IMAGES_ERROR
})

export const getMovieCredits = movieId => ({
  type: GET_MOVIE_CREDITS,
  movieId
})

export const requestMovieCreditsStart = () => ({
  type: REQUEST_MOVIE_CREDITS_START
})
export const requestMovieCreditsError = () => ({
  type: REQUEST_MOVIE_CREDITS_ERROR
})

export const requestMovieCreditsSuccess = (cast, crew) => ({
  type: REQUEST_MOVIE_CREDITS_SUCCESS,
  cast,
  crew
})
