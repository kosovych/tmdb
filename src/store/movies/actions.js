import { ADD_MOVIES } from './types'

export const addMovies = (payload, endpoint) => ({
  type: ADD_MOVIES,
  payload,
  endpoint
})
