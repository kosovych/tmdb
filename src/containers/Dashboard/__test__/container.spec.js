import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import diveTo from 'Utils/testHelpers/diveToEnzyme'
import { getMovies as getMoviesAction } from 'Store/trendingMovies/actions'
import DashboardWrapper, { DashboardContainer } from '../container'

jest.mock('Store/trendingMovies/actions', () => ({
  getMovies: jest.fn(),
  setSearch: jest.fn()
}))

describe('Login Container', () => {
  const store = configureStore()({})
  store.dispatch = jest.fn()
  const wrapper = shallow(<DashboardWrapper store={store} />)
  const container = diveTo(wrapper, DashboardContainer)

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot()
  })

  describe('componentDidMount()', () => {
    it('should call getMoviesAction() action', () => {
      const instance = container.instance()
      instance.componentDidMount()
      expect(getMoviesAction).toHaveBeenCalledTimes(1)
    })
  })

  describe('onPageChange()', () => {
    const page = { page: 1 }
    const instance = container.instance()
    it('should call getMoviesAction() action with right params', () => {
      instance.onPageChange(page)
      expect(getMoviesAction).toHaveBeenCalledTimes(1)
    })
  })
})
