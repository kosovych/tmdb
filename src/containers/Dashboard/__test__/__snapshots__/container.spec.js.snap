// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container matches snapshot 1`] = `
<Dashboard
  currentPage={null}
  error={null}
  isBlank={true}
  loading={null}
  movies={Array []}
  onPageChange={[Function]}
  totalResults={null}
/>
`;
